app/models/article.rb

```diff
@@ -73,4 +77,10 @@ def generate_published_dates
   def downcase_content_format
     self.content_format = self.content_format.downcase
   end
+
+  def create_redirect
+    unless Redirect.exists?(source_path: short_path, target_path: path)
+      Redirect.create(source_path: short_path, target_path: path)
+    end
+  end
```

Comment:
I'm looking at this right now, and probably wanting to wrap it up tomorrow or Wednesday evening if that works.